### PR TITLE
Add setAppUserId in PropertyConfigurator

### PR DESCRIPTION
**Pull request by guperrot**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Add setAppUserId in PropertyConfigurator.

## Related PRs or issues

#1228

### CODE REVIEW

File: AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSPropertyConfigurator.m [hunk 1]
```diff
@@ -46,6 +47,26 @@ - (void)setAppLocale:(NSString *)appLocale {
   _appLocale = appLocale;
 }
 
+- (void)setAppUserId:(NSString *)appUserId {
+  @synchronized([MSAnalytics sharedInstance]) {
```
Comment (jaelim-ms) [inline AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSPropertyConfigurator.m]: Any reason for synchronization? If we need this then we need to synchronize other methods.

File: AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTransmissionTargetTests.m [hunk 2]
```diff
@@ -1140,15 +1160,81 @@ - (void)testOverridingCommonSchemaPropertiesWithTwoChildrenUnderTheSameParent {
...
+  log.ext.appExt = [MSAppExtension new];
+
+  // When
+  [target.propertyConfigurator channel:nil prepareLog:log];
+
+  // Then
+  XCTAssertNil(log.ext.appExt.userId);
+}
+
+- (void)testOverridingValidAppUserIdThenUnset {
```
Comment (dhei) [inline AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTransmissionTargetTests.m]: In addition to this test case, can we add a test case where overriding with valid AppUserId then overriding with invalid AppUserId to make sure that the second AppUserId doesn't override the first AppUserId?

File: AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSPropertyConfigurator.h [hunk 3]
```diff
@@ -25,6 +25,14 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (void)setAppLocale:(nullable NSString *)appLocale;
 
+/**
+ * Set the user identifier.
+ * The user identifier needs to start with c: or i: or d: or w: prefixes.
```
Comment (clpolet) [inline AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSPropertyConfigurator.h]: Try to fit everything in one line as much as possible otherwise the doc may get truncated by Xcode. To try it out keep the "option/alt" key pressed while clicking the method, Xcode will compute  the documentation.

File: AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSPropertyConfigurator.m (hunk 1 above)
Comment (guperrot) [reply AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSPropertyConfigurator.m]: The check is not atomic with the update (if). We also do that in other existing methods that have more than just a variable assignment.

File: AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSPropertyConfigurator.h (hunk 3 above)
Comment (guperrot) [reply AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSPropertyConfigurator.h]: screen shot 2018-11-20 at 1 30 33 pm

To me it looks good with the current, not sure what you are asking?

Comment (clpolet) [reply AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSPropertyConfigurator.h]: Let me try again, not sure which version of Xcode didn't work.

File: AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTransmissionTargetTests.m (hunk 2 above)
Comment (guperrot) [reply AppCenterAnalytics/AppCenterAnalyticsTests/MSAnalyticsTransmissionTargetTests.m]: Done in https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1229/commits/30851986d2b5b4679249622a2ad729fe948c0a4c

File: AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSPropertyConfigurator.m (hunk 1 above)
Comment (guperrot) [reply AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSPropertyConfigurator.m]: Also it creates the static cached regex only once

File: AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSPropertyConfigurator.h (hunk 3 above)
Comment (clpolet) [reply AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSPropertyConfigurator.h]: Nevermind, that was Xcode 8 picking up the doc from the property instead of the setter :man_facepalming:. Xcode 10.1 takes the right doc. 
screen shot 2018-11-20 at 13 36 34
